pkg/wasm/main.go

```diff
@@ -0,0 +1,212 @@
+//go:build wasm
+// +build wasm
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"syscall/js"
+
+	"github.com/authzed/spicedb/pkg/datastore"
+	"github.com/authzed/spicedb/pkg/development"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+	devinterface "github.com/authzed/spicedb/pkg/proto/developer/v1"
+	"github.com/authzed/spicedb/pkg/schemadsl/compiler"
+	"github.com/authzed/spicedb/pkg/schemadsl/generator"
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gookit/color"
+	"github.com/jzelinskie/cobrautil/v2"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	"github.com/authzed/zed/internal/client"
+	"github.com/authzed/zed/internal/commands"
+	"github.com/authzed/zed/internal/console"
+)
+
+// zedCommandResult is the struct JSON serialized to be returned to the caller.
+type zedCommandResult struct {
+	UpdatedContext string `json:"updated_context"`
+	Output         string `json:"output"`
+	Error          string `json:"error"`
+}
+
+func main() {
```

Comment:
Would it not be possible to have as much shared as possible and then only a little wrapper that sets things up in `cmd/zed/main_wasm.go` and `cmd/zed/main.go`?

Reply:
This does share as much as I believed I could make shared; the command implementations are all shared, but the actual processing needs to be custom for the JS calling interface